src/search.c

```diff
@@ -5255,6 +5256,8 @@ search_for_fuzzy_match(
 			*pos = current_pos;
 			break;
 		    }
+		    else if (looped_around && current_pos.lnum == circly_end.lnum)
```

Comment:
Isn't this caught in line 5237?

Reply:
Nope. like when startpos is lnum 1, col 5, in first running the ptr will from col 5 ` *ptr += current_pos.col;` we search from the col 5 . then loop back to lnum 1 currentpos.col is 0 because it will reset to 0 . then use ptr do fuzzy if this line nothing matched. should break. but in this case currentpos is lnum 1, col 0. so 5237 can't caught it in EQUAL_POS.